main/src/library/File.flix

```diff
@@ -24,22 +88,58 @@ namespace File {
             import java.io.File.canRead();
             Ok(canRead(newFile(f)))
         } catch {
-            case _: ##java.io.IOException => Err("I/O Error")
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns `true` if the given file `f` is writable.
+    ///
+    pub def isWritable(f: String): Result[Bool, String] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.canWrite();
+            Ok(canWrite(newFile(f)))
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns `true` if the given file `f` is executable.
+    ///
+    pub def isExecutable(f: String): Result[Bool, String] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.canExecute();
+            Ok(canExecute(newFile(f)))
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
         }
 
     ///
     /// Returns a list of all the lines in the given file `f`.
     ///
-    pub def readLines(f: String): Result[List[String], String] & Impure = try {
+    pub def readLines(f: String): Result[List[String], String] & Impure =
+        try {
             import new java.io.File(String) as newFile;
             import java.io.File.toPath();
             import java.nio.file.Files:newBufferedReader(##java.nio.file.Path);
+            import java.io.BufferedReader.close();
             let javaFile = newFile(f);
             let javaPath = toPath(javaFile);
             let reader = newBufferedReader(javaPath);
-            Ok(readAll(reader))
+            let result = readAll(reader);
+            close(reader);
+            Ok(result)
         } catch {
-            case _: ##java.io.IOException => Err("I/O Error")
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
         }
 
     ///
```

Comment:
Is there a reason to prefer this instead of using [this java method](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Files.html#readAllLines(java.nio.file.Path,java.nio.charset.Charset))

Reply:
A reason is that it returns a Java list and not a Flix list (and there is no good way to convert one into the other).